src/core/xfa/fonts.js

```diff
@@ -0,0 +1,161 @@
+        font = Object.create(null);
+        this.fonts.set(name, font);
+        if (!this.defaultFont) {
+          this.defaultFont = font;
+        }
+      }
+      let property = "";
+      if (cssFontInfo.italicAngle !== "0") {
+        if (parseFloat(cssFontInfo.fontWeight) >= 700) {
+          property = "bolditalic";
+        } else {
+          property = "italic";
+        }
+      } else if (parseFloat(cssFontInfo.fontWeight) >= 700) {
+        property = "bold";
+      }
+
+      if (!property) {
+        if (
+          pdfFont.name.includes("Bold") ||
+          (pdfFont.psName && pdfFont.psName.includes("Bold"))
+        ) {
+          property = "bold";
+        }
+        if (
+          pdfFont.name.includes("Italic") ||
+          pdfFont.name.endsWith("It") ||
+          (pdfFont.psName &&
+            (pdfFont.psName.includes("Italic") ||
+              pdfFont.psName.endsWith("It")))
+        ) {
+          property += "italic";
+        }
+      }
+
+      if (!property) {
+        property = "regular";
+      }
+
+      font[property] = pdfFont;
+    }
+
+    for (const pdfFont of this.fonts.values()) {
+      if (!pdfFont.regular) {
+        pdfFont.regular = pdfFont.italic || pdfFont.bold || pdfFont.bolditalic;
+      }
+    }
+  }
+
+  getDefault() {
+    return this.defaultFont;
+  }
+
+  find(fontName, mustWarn = true) {
+    let font = this.fonts.get(fontName) || this.cache.get(fontName);
+    if (font) {
+      return font;
+    }
+
+    const pattern = /,|-| |bolditalic|bold|italic|regular|it/gi;
+    let name = fontName.replace(pattern, "");
+    font = this.fonts.get(name);
+    if (font) {
+      this.cache.set(fontName, font);
+      return font;
+    }
+    name = name.toLowerCase();
+
+    const maybe = [];
+    for (const [family, pdfFont] of this.fonts.entries()) {
+      if (family.replace(pattern, "").toLowerCase().startsWith(name)) {
+        maybe.push(pdfFont);
+      }
+    }
+
+    if (maybe.length === 0) {
+      for (const [, pdfFont] of this.fonts.entries()) {
+        if (
+          pdfFont.regular.name &&
+          pdfFont.regular.name
+            .replace(pattern, "")
+            .toLowerCase()
+            .startsWith(name)
+        ) {
+          maybe.push(pdfFont);
+        }
+      }
+    }
+
+    if (maybe.length === 0) {
+      name = name.replace(/psmt|mt/gi, "");
+      for (const [family, pdfFont] of this.fonts.entries()) {
+        if (family.replace(pattern, "").toLowerCase().startsWith(name)) {
+          maybe.push(pdfFont);
+        }
+      }
+    }
+
+    if (maybe.length === 0) {
+      for (const pdfFont of this.fonts.values()) {
+        if (
+          pdfFont.regular.name &&
+          pdfFont.regular.name
+            .replace(pattern, "")
+            .toLowerCase()
+            .startsWith(name)
+        ) {
+          maybe.push(pdfFont);
+        }
+      }
+    }
+
+    if (maybe.length >= 1) {
+      if (maybe.length !== 1) {
+        if (mustWarn) {
```

Comment:
I'm guessing that *one* if, i.e. `if (maybe.length !== 1 && mustWarn) {`, would suffice here?